paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonScanBuilder.scala

```diff
@@ -33,4 +40,45 @@ class PaimonScanBuilder(table: Table)
     // just make a best effort to push down limit
     false
   }
+
+  override def supportCompletePushDown(aggregation: Aggregation): Boolean = {
+    // for now we only support complete push down, so there is no difference with `pushAggregation`
+    pushAggregation(aggregation)
+  }
+
+  // Spark does not support push down aggregation for streaming scan.
+  override def pushAggregation(aggregation: Aggregation): Boolean = {
+    if (localScan.isDefined) {
+      return true
+    }
+
+    // Only support with push down partition filter
+    if (pushed.length != partitionFilter.length) {
+      return false
+    }
+
+    val aggregator = new LocalAggregator(table)
+    if (!aggregator.supportAggregation(aggregation)) {
+      return false
+    }
+
+    val readBuilder = table.newReadBuilder
+    if (pushed.nonEmpty) {
```

Comment:
just use partitionFilter?

Reply:
`partitionFilters` is the Spark filter, but we need Paimon's predicate..